### PR TITLE
Replace internal esm implementation with esm2quinox

### PR DESCRIPTION
**Pull request by samtomguns**
Replace the built-in ESM-2 equinox implementation with `esm2quinox` from https://github.com/patrick-kidger/esm2quinox/tree/main. 

There's some minor (10 lines) code duplication of the transformer trunk because `esm2quinox` only has a combined trunk+head function. 

I've checked that the pseudo-LL and gradients reproduce (up to floating point errors) for random inputs.

### CODE REVIEW

File: src/boltz_binder_design/losses/esm.py [hunk 1]
```diff
@@ -33,21 +46,22 @@ def __call__(self, seq_standard_tokens: Float[Array, "N 20"], *, key):
...
             # set masked token embedding to zero
             embedding = embedding.at[index].set(0.0)
             # rescale to account for masking during ESM training
             mask_ratio_train = 0.15 * 0.8
             embedding = embedding * ((1 - mask_ratio_train) / (1 - 1/(n+2)))
             # apply ESM trunk and LM head
-            embedding = self.esm._apply_trunk(embedding, np.ones((n + 2, n + 2)))
-            return jax.nn.log_softmax(self.esm.lm_head(embedding))[index]
+            embedding = self._apply_trunk(embedding, np.zeros(n + 2))
+            return jax.nn.log_softmax(jax.vmap(self.esm.logit_head)(embedding))[index]
```
Comment (nboyd) [inline src/boltz_binder_design/losses/esm.py:64]: Super minor and I probably did this too, but you could probably replace this with `jax.nn.log_softmax(self.esm.logit_head(embedding[index]))`

File: src/boltz_binder_design/losses/esm.py [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
+        esm_idx = ESM_TOKENS[tok]
         T[i, esm_idx] = 1
     return T
 
-
 class ESM2PseudoLikelihood(LossTerm):
     esm: ESM2
     stop_grad: bool = True
 
+    @eqx.filter_jit #necessary ?
```
Comment (nboyd) [inline src/boltz_binder_design/losses/esm.py:24]: nope! this should all be under one top-level JIT (though the `jax.lax.scan` means the meat of this function would be JIT'd either way AFAICT)

**Review by nboyd — APPROVED** [review]
Nice work!

File: src/boltz_binder_design/esm/esm.py [hunk 3]
```diff
@@ -1,184 +1,41 @@
...
-        mask_ratio_observed = (tokens == MASK_IDX).sum() / src_length
-        # Rescale by the ratio between theoretical and observed.
-        embeds = embeds * (1 - mask_ratio_train) / (1 - mask_ratio_observed)
-
-        return embeds
+class ESM2(object):
+    esm: esm2quinox.ESM2
+
+    def __init__(self, *args, model=None, **kwargs):
+        models = {
```
Comment (nboyd) [inline src/boltz_binder_design/esm/esm.py:15]: I really don't like this -- without reading this code it's impossible to tell what values "model" can take. Much better IMO to just take a torch model as an arg; if you want give an example of how to load one in the doc string.

File: src/boltz_binder_design/esm/esm.py [hunk 4]
```diff
@@ -1,184 +1,41 @@
...
-
-        mask_ratio_train = 0.15 * 0.8
-        src_length = (tokens != PAD_IDX).sum() + 1e-8
-        # Get the mask token to sequence length ratio
-        mask_ratio_observed = (tokens == MASK_IDX).sum() / src_length
-        # Rescale by the ratio between theoretical and observed.
-        embeds = embeds * (1 - mask_ratio_train) / (1 - mask_ratio_observed)
-
-        return embeds
+class ESM2(object):
```
Comment (nboyd) [inline src/boltz_binder_design/esm/esm.py:6]: Probably better to make this an equinox.Module -- if we're going down the wrapper route you'll want to include this in various loss functions

**Review by nboyd — CHANGES_REQUESTED** [review]

File: src/boltz_binder_design/esm/esm.py [hunk 5]
```diff
@@ -24,12 +29,12 @@ def __init__(self, *args, model=None, **kwargs):
         else:
             self.esm = esm2quinox.ESM2(*args, **kwargs)
 
+    ### encapsulate wrapped class 
     def __getattr__(self, attr):
```
Comment (nboyd) [inline src/boltz_binder_design/esm/esm.py:33]: Please no

File: src/boltz_binder_design/losses/esm.py (hunk 2 above)
Comment (samtomguns) [reply src/boltz_binder_design/losses/esm.py]: thanks, removed

File: src/boltz_binder_design/esm/esm.py (hunk 4 above)
Comment (samtomguns) [reply src/boltz_binder_design/esm/esm.py]: fixed, thanks
